### DISCUSSION TITLE
Graphql support?

### DISCUSSION THREAD

**Discussion by madarco** [marcj/deepkit#259] [Ideas]
Saw the reddit post about this framework and I think it had a huge potential!
Any plan for a Graphql support?

**Comment by marcus-sa** [comment]
~~I think you'd have to write that implementation yourself.
There's no point in using GraphQL with this framework when we have the RPC library.~~
You can find it [here](https://github.com/marcus-sa/deepkit-graphql).

**Comment by madarco** [comment]
Thanks for the prompt reply.

I see, maybe could be a nice side project.
I think there are still many benefits to using Graphql, like not tying the frontend to this framework, avoiding BFF pattern, fetching only required data, addressing N+1 problems, etc.

**Reply by marcj** [reply]
Yeah, I see a bit the BFF point but not the others as RPC solves that too. I'm not a fan of GraphQL due to its custom DSL. I hate to repeat my types in a less poweful language like that, so I avoid it where possible and don't want to invest time into integrating it. I'd welcome a community library though and certainly help where necessary to make it possible. So if you or someone else gives it a try, let me know.

**Reply by madarco** [reply]
Yes me too! I hate to have to repeat the types. 

Code-first solutions like https://typegraphql.com/ alleviate the problem, but without runtime type information they are quite limited. A problem that you solved brilliantly :)

Also, @deepkit/type could make jit possible for serialization and query parsing, like https://www.npmjs.com/package/graphql-jit or mercurius.

**Comment by WonderPanda** [comment]
I think that the best course of action would be for a community module to be implemented using [GraphQL Yoga 2](https://www.the-guild.dev/blog/announcing-graphql-yoga-v2) as its designed with a pluggable architecture in mind

**Comment by hanayashiki** [comment]
graphql is a too radical openapi. Maybe considering the common requirments of fetching only required data we should not just build a new language

**Comment by marcus-sa** [comment]
@madarco  I've created a GraphQL integration for Deepkit https://github.com/marcus-sa/deepkit-graphql

**Comment by seivan** [comment]
I mean wouldn't it just be enough to take the request body as a certain path and pass it over to something like Nexus or Yoga. 
GraphQL is supposed to be transport agnostic, so there isn't really something that needs to be done here. 
That being said, certain transport like websockets would need deeper integration, but I am not a 100%.

**Reply by marcus-sa** [reply]
Nope, not if we want to utilize TypeScript to the fullest
https://github.com/marcus-sa/deepkit-graphql
